patches/net/minecraft/world/item/crafting/Ingredient.java.patch

```diff
@@ -178,6 +178,19 @@
  
          @Override
          public boolean equals(Object p_301316_) {
+@@ -156,6 +_,12 @@
+                 : ingredient$itemvalue.item.getItem().equals(this.item.getItem()) && ingredient$itemvalue.item.getCount() == this.item.getCount();
+         }
+ 
++        // Neo: Add a hashCode() implementation that matches equals() to avoid violating contract
```

Comment:
Beyond the contract, the problem is modder expectations. I think that we can omit "to avoid violating the contract".